Fix crash when viewing audio levels in Scene Cast

manongjohn opened the pull request:
This PR fixes #1240 

The "View" menu item does not support Audio and Palette level types.  Disabled this option if either of those types are selected to prevent crashing.

Additionally allowed "Remove Level" and "Remove All Unused Levels" to be options when selecting Audio level types as these are still valid actions.

turtletooth commented on toonz/sources/toonz/castviewer.cpp:
```diff
@@ -752,11 +752,12 @@ QMenu *CastBrowser::getContextMenu(QWidget *parent, int index) {
-    return menu;
+//  if (audioSelected && !paletteSelected && !vectorLevelSelected &&
+//      !otherFileSelected)
+//    return menu;
```
Thanks for this.  I think these commented lines can be fully removed.

manongjohn replied:
Done. Thanks for reviewing.

turtletooth commented:
@manongjohn The crash can still happen via keyboard shortcut.

adding this to levelsettingspopup.cpp seems to solve it:

`for (i = 0; i < selectedLevels.size(); i++) {
		  if (selectedLevels[i]->getType() == SND_XSHLEVEL) return;
		  simpleLevels.push_back(selectedLevels[i]->getSimpleLevel());
	  }`

manongjohn commented:
@turtletooth Thanks for reviewing and finding the issue with the View shortcut.

I've corrected the logic to make sure it doesn't try to work on levels it shouldn't.  Really, the only valid level types that View should work on are: Toonz Vector, Toonz Raster and standard Raster.

Please note, I also fixed the Scene Cast windows so View doesn't work on Mesh levels.  The option is not usually allowed in the Xsheet context menu for this type.  Viewing a mesh level doesn't cause an immediate crash but I was sporadically crashing every time I closed the flipbook that opened for the mesh level.

turtletooth commented:
LGTM.  Thanks @manongjohn
